src/net/if_.rs

```diff
@@ -0,0 +1,48 @@
+//! Network interface name resolution.
+//!
+//! Uses Linux and/or POSIX functions to resolve interface names like "eth0"
+//! or "socan1" into device numbers.
+
+use libc::{c_char, c_uint};
+use std::ffi::{CString, NulError};
+use std::io;
+
+/// Error that can occur during interface name resolution.
+#[derive(Debug)]
+pub enum NameToIndexError {
+    /// Failed to allocate a C-style string to for the syscall
+    NulError,
+    IOError(io::Error),
+}
```

Comment:
I looked into it before implementing my own, but there's the NulError to take care of. I believe it may be worthwhile to add that to nix::Error though? It's documented in https://doc.rust-lang.org/std/ffi/struct.NulError.html.

Alternatives are just returning not found if a NulError occured or, as a last resort, unpacking (I'd like to avoid that).